### PR TITLE
[frontend] Fixes undefined method `to_datetime' for nil:NilClass

### LINKED ISSUE #5045: [production][] undefined method `to_datetime' for nil:NilClass
threw this 148 times
<pre>
undefined method `to_datetime&#39; for nil:NilClass
</pre>

[See this exception on Errbit](https://errbit-opensuse.herokuapp.com/apps/5620ca2bdc71fa00b1000000/problems/5aa298ae2e809000065e2c82)

### PR DESCRIPTION
**Pull request by bgeuken**
When a binary doesn't have a buildtime OBS was causing an exception,
because we were not handling this case.
This fixes the issue.

Fixes #5045

### CODE REVIEW

File: src/api/app/models/binary_release.rb [hunk 1]
```diff
@@ -57,7 +57,7 @@ def self.update_binary_releases_via_json(repository, json, time = Time.now)
           entry = existing.first
           if entry.binary_disturl                   == binary['disturl'] &&
              entry.binary_supportstatus             == binary['supportstatus'] &&
-             entry.binary_buildtime.to_datetime.utc == ::Time.at(binary['buildtime'].to_i).to_datetime.utc
+             entry.binary_buildtime.to_datetime.utc == ::Time.at(binary['buildtime'].to_i).try(:to_datetime).try(:utc)
```
Comment (Ana06) [inline src/api/app/models/binary_release.rb]: how can `at(binary['buildtime'].to_i)` return `nil`? Why is `::Time` written with `::`? Is this `Time` class the [default Ruby one](https://ruby-doc.org/core-2.5.0/Time.html#method-c-at)? :confused:

Comment (Ana06) [reply src/api/app/models/binary_release.rb]: I think I know it! :tada: What can be `nil` is in the other part of the `==`. `entry.binary_buildtime` can be `nil`, `Time.at(binary['buildtime'].to_i)` can never be `nil` :nerd_face: So this is not solving the bug :rofl: 

You could do for example:

```ruby
entry.binary_buildtime && entry.binary_buildtime.to_datetime.utc == Time.at(binary['buildtime'].to_i).to_datetime.utc

Comment (bgeuken) [reply src/api/app/models/binary_release.rb]: Oh, right :flushed:

**Comment by bgeuken** [comment]
@Ana06 Fixed. I have some follow up refactorings that I'll submit in a separate PR

File: src/api/app/models/binary_release.rb [hunk 2]
```diff
@@ -184,6 +182,12 @@ def reset_cache
     Rails.cache.delete("xml_binary_release_#{cache_key}")
   end
 
+  def indentical_to?(binary_hash)
+    binary_disturl == binary_hash['disturl'] &&
+      binary_supportstatus == binary_hash['supportstatus'] &&
+      binary_buildtime.present? &&
```
Comment (Ana06) [inline src/api/app/models/binary_release.rb]: `present?` is not needed, `build_buildtime` is either a time or `nil`, so you can remove the `present` as it is in my [previous comment](https://github.com/openSUSE/open-build-service/pull/5149#discussion_r196119971) :wink:

**Pull request merged**